Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1539 @@
+
+    ScopedFileHandle fileResource("check_nxsource_group_test_file.hdf5");
+    std::string destinationFile = fileResource.fullPath();
+
+    auto const &compInfo = (*m_instrument.first);
+
+    NexusGeometrySave::saveInstrument(m_instrument, destinationFile,
+                                      DEFAULT_ROOT_PATH);
+    HDF5FileTestUtility tester(destinationFile);
+    TS_ASSERT(compInfo.hasSource());
+    TS_ASSERT(tester.parentNXgroupHasChildNXgroup(NX_INSTRUMENT, NX_SOURCE));
+  }
+
+  void test_nxsample_group_exists_and_is_in_nxentry_group() {
+
+    ScopedFileHandle fileResource("check_nxsource_group_test_file.hdf5");
+    std::string destinationFile = fileResource.fullPath();
+
+    auto const &compInfo = (*m_instrument.first);
+
+    NexusGeometrySave::saveInstrument(m_instrument, destinationFile,
+                                      DEFAULT_ROOT_PATH);
+    HDF5FileTestUtility tester(destinationFile);
+
+    TS_ASSERT(compInfo.hasSample());
+    TS_ASSERT(tester.parentNXgroupHasChildNXgroup(NX_ENTRY, NX_SAMPLE));
+  }
+
+  /*
+====================================================================
+
+NEXUS TRANSFOMATIONS TESTS
+
+DESCRIPTION:
+
+The following tests document that saveInstrument will find and write detectors
+and other Instrument components to file in Nexus format, and where there exists
+transformations in ComponentInfo and DetectorInfo, SaveInstrument will generate
+'NXtransformations' groups to contain the corresponding component
+rotations/translations, and pixel offsets in any 'NXdetector' or 'NXmonitor'
+found in the Instrument cache.
+
+LIST(S):
+
+* test when NXdetector groups have NXtransformations, 'transformation_type' is
+specified for all
+* test when NXmonitor groups have NXtransformations, 'transformation_type' is
+specified for all
+*  test when NXsource group has NXtransformations, 'transformation_type' is
+specified
+* test x/y/z pixel offset in file is relative position from bank without
+rotation
+
+* test rotation of NXdetector written to file is same as in component info.
+* test rotation of NXmonitor written to file is same as in component info.
+* test rotation of source written to file is same as in component info.
+
+LOCATION TESTS:
+* test NXmonitor location not written when is at origin
+* test NXdetector location not written when is at origin
+* test NXsource location not written when is at origin
+
+ROTATION TESTS
+* test NXdetector rotation not written when is zero
+* test NXmonitor rotation not written when is zero
+* test NXsource rotation not written when is zero
+
+====================================================================
+*/
+
+  void
+  test_when_nx_detector_groups_have_nx_transformations_transformation_type_is_specified_for_all() {
+
+    ScopedFileHandle fileResource(
+        "check_nxdetector_groups_have_transformation_types_test_file.hdf5");
+    std::string destinationFile = fileResource.fullPath();
+
+    auto instrument =
```

Comment:
For all test methods creating instruments (1) Comment on why you need an instrument of this type. (2) provide inline comments `/**/` for parameters passed for improved readability.